pkg/controller/release/strategy_executor.go

```diff
@@ -147,21 +149,47 @@ func (e *StrategyExecutor) Execute(prev, curr, succ *releaseInfo) (bool, []Strat
 	pipeline := NewPipeline()
 	pipeline.Enqueue(genInstallationEnforcer(ctx, curr, succ))
 
-	if isHead {
-		pipeline.Enqueue(genCapacityEnforcer(ctx, curr, succ))
-		pipeline.Enqueue(genTrafficEnforcer(ctx, curr, succ))
-		if hasTail {
-			// This is the moment where a contender is performing a look-behind.
-			// Incumbent's context is completely identical to it's successor
-			// except that it's not the head of the chain anymore.
-			prevctx := ctx.Copy()
-			prevctx.isHead = false
-			pipeline.Enqueue(genTrafficEnforcer(prevctx, prev, curr))
-			pipeline.Enqueue(genCapacityEnforcer(prevctx, prev, curr))
+	if e.progressing {
+		// progressing release (target step > achieved step):
+			// increase capacity for current release
+			// increase traffic for current release
+			// reduce traffic for previous release
+			// reduce capacity for previous release
+		if isHead {
+			pipeline.Enqueue(genCapacityEnforcer(ctx, curr, succ))
+			pipeline.Enqueue(genTrafficEnforcer(ctx, curr, succ))
+			if hasTail {
+				// This is the moment where a contender is performing a look-behind.
+				// Incumbent's context is completely identical to it's successor
+				// except that it's not the head of the chain anymore.
+				prevctx := ctx.Copy()
+				prevctx.isHead = false
+				pipeline.Enqueue(genTrafficEnforcer(prevctx, prev, curr))
+				pipeline.Enqueue(genCapacityEnforcer(prevctx, prev, curr))
+			}
+		} else {
+			pipeline.Enqueue(genTrafficEnforcer(ctx, curr, succ))
+			pipeline.Enqueue(genCapacityEnforcer(ctx, curr, succ))
 		}
 	} else {
-		pipeline.Enqueue(genTrafficEnforcer(ctx, curr, succ))
-		pipeline.Enqueue(genCapacityEnforcer(ctx, curr, succ))
+		// release is not progressing (target step > achieved step):
```

Comment:
You probably meant: "// release is not progressing (target step < achieved step)"